<a href="https://colab.research.google.com/github/RohitJaiswal01/Python-EDA-Air-BNB-2019-Analysis/blob/main/Air_BNB_2019_V2_EDA_Submission_Template.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# **Project Name**    -



##### **Project Type**    - EDA
##### **Contribution**    - Individual
##### **Team Member 1 -**
##### **Team Member 2 -**
##### **Team Member 3 -**
##### **Team Member 4 -**

# **Project Summary -**

Write the summary here within 500-600 words.
Story of what you have done in the whole project not like conclusion of

# **GitHub Link -**

**Python-EDA-Air-BNB-2019-Analysis repository**

https://github.com/RohitJaiswal01/Python-EDA-Air-BNB-2019-Analysis

# **Problem Statement**


**Air BNB Dataset Analysis**

Since 2008, guests and hosts have used Airbnb to expand on travelling possibilities and present a more unique, personalised way of experiencing the world. Today, Airbnb became one of a kind service that is used and recognized by the whole world. Data analysis on millions of listings provided through Airbnb is a crucial factor for the company. These millions of listings generate a lot of data - data that can be analysed and used for security, business decisions, understanding of customers' and providers' (hosts) behaviour and performance on the platform, guiding marketing initiatives, implementation of innovative additional services and much more. This dataset has around 49,000 observations in it with 16 columns and it is a mix of categorical and numeric values. Explore and analyse the data to discover key understandings.

#### **Define Your Business Objective?**

Analysis and generating insights from the data, helping in understanding of customers' and providers' (hosts) behaviour and performance on the platform, guiding marketing initiatives, implementation of secure and innovative additional services and much more

Data oriented business decisions, helps in improving processes and services, translating the complex data and discover key understanding for future actionable plans.

# **General Guidelines** : -  

1.   Well-structured, formatted, and commented code is required.
2.   Exception Handling, Production Grade Code & Deployment Ready Code will be a plus. Those students will be awarded some additional credits.
     
     The additional credits will have advantages over other students during Star Student selection.
       
             [ Note: - Deployment Ready Code is defined as, the whole .ipynb notebook should be executable in one go
                       without a single error logged. ]

3.   Each and every logic should have proper comments.
4. You may add as many number of charts you want. Make Sure for each and every chart the following format should be answered.
        

```
# Chart visualization code
```
            

*   Why did you pick the specific chart?
*   What is/are the insight(s) found from the chart?
* Will the gained insights help creating a positive business impact?
Are there any insights that lead to negative growth? Justify with specific reason.

5. You have to create at least 20 logical & meaningful charts having important insights.


[ Hints : - Do the Vizualization in  a structured way while following "UBM" Rule.

U - Univariate Analysis,

B - Bivariate Analysis (Numerical - Categorical, Numerical - Numerical, Categorical - Categorical)

M - Multivariate Analysis
 ]





# ***Let's Begin !***

## ***1. Know Your Data***

### Import Libraries

In [ ]:
# Import Libraries
import numpy as np
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt


### Dataset Loading

In [ ]:
# Load Dataset
# Loading the repository files from the github to collab
!git clone https://github.com/RohitJaiswal01/Python-EDA-Air-BNB-2019-Analysis.git

# Giving a path to the file
path = "/content/Python-EDA-Air-BNB-2019-Analysis/Airbnb NYC 2019.csv"

Cloning into 'Python-EDA-Air-BNB-2019-Analysis'...
remote: Enumerating objects: 6, done.
remote: Counting objects: 100% (6/6), done.
remote: Compressing objects: 100% (5/5), done.
remote: Total 6 (delta 0), reused 0 (delta 0), pack-reused 0
Receiving objects: 100% (6/6), 2.25 MiB | 10.78 MiB/s, done.


In [ ]:
# Read/load the dataset from the loaded repository files
dataset = pd.read_csv(path)
dataset

id                                               name   host_id  \
0          2539                 Clean & quiet apt home by the park      2787   
1          2595                              Skylit Midtown Castle      2845   
2          3647                THE VILLAGE OF HARLEM....NEW YORK !      4632   
3          3831                    Cozy Entire Floor of Brownstone      4869   
4          5022   Entire Apt: Spacious Studio/Loft by central park      7192   
...         ...                                                ...       ...   
48890  36484665    Charming one bedroom - newly renovated rowhouse   8232441   
48891  36485057      Affordable room in Bushwick/East Williamsburg   6570630   
48892  36485431            Sunny Studio at Historical Neighborhood  23492952   
48893  36485609               43rd St. Time Square-cozy single bed  30985759   
48894  36487245  Trendy duplex in the very heart of Hell's Kitchen  68119814   

           host_name neighbourhood_group       neighbourhood  latitude  \
0               John            Brooklyn          Kensington  40.64749   
1           Jennifer           Manhattan             Midtown  40.75362   
2          Elisabeth           Manhattan              Harlem  40.80902   
3        LisaRoxanne            Brooklyn        Clinton Hill  40.68514   
4              Laura           Manhattan         East Harlem  40.79851   
...              ...                 ...                 ...       ...   
48890        Sabrina            Brooklyn  Bedford-Stuyvesant  40.67853   
48891        Marisol            Brooklyn            Bushwick  40.70184   
48892  Ilgar & Aysel           Manhattan              Harlem  40.81475   
48893            Taz           Manhattan      Hell's Kitchen  40.75751   
48894     Christophe           Manhattan      Hell's Kitchen  40.76404   

       longitude        room_type  price  minimum_nights  number_of_reviews  \
0      -73.97237     Private room    149               1                  9   
1      -73.98377  Entire home/apt    225               1                 45   
2      -73.94190     Private room    150               3                  0   
3      -73.95976  Entire home/apt     89               1                270   
4      -73.94399  Entire home/apt     80              10                  9   
...          ...              ...    ...             ...                ...   
48890  -73.94995     Private room     70               2                  0   
48891  -73.93317     Private room     40               4                  0   
48892  -73.94867  Entire home/apt    115              10                  0   
48893  -73.99112      Shared room     55               1                  0   
48894  -73.98933     Private room     90               7                  0   

      last_review  reviews_per_month  calculated_host_listings_count  \
0      2018-10-19               0.21                               6   
1      2019-05-21               0.38                               2   
2             NaN                NaN                               1   
3      2019-07-05               4.64                               1   
4      2018-11-19               0.10                               1   
...           ...                ...                             ...   
48890         NaN                NaN                               2   
48891         NaN                NaN                               2   
48892         NaN                NaN                               1   
48893         NaN                NaN                               6   
48894         NaN                NaN                               1   

       availability_365  
0                   365  
1                   355  
2                   365  
3                   194  
4                     0  
...                 ...  
48890                 9  
48891                36  
48892                27  
48893                 2  
48894                23  

[48895 rows x 16 columns]

### Dataset First View

In [ ]:
# Dataset First Look
dataset.head()

id                                              name  host_id  \
0  2539                Clean & quiet apt home by the park     2787   
1  2595                             Skylit Midtown Castle     2845   
2  3647               THE VILLAGE OF HARLEM....NEW YORK !     4632   
3  3831                   Cozy Entire Floor of Brownstone     4869   
4  5022  Entire Apt: Spacious Studio/Loft by central park     7192   

     host_name neighbourhood_group neighbourhood  latitude  longitude  \
0         John            Brooklyn    Kensington  40.64749  -73.97237   
1     Jennifer           Manhattan       Midtown  40.75362  -73.98377   
2    Elisabeth           Manhattan        Harlem  40.80902  -73.94190   
3  LisaRoxanne            Brooklyn  Clinton Hill  40.68514  -73.95976   
4        Laura           Manhattan   East Harlem  40.79851  -73.94399   

         room_type  price  minimum_nights  number_of_reviews last_review  \
0     Private room    149               1                  9  2018-10-19   
1  Entire home/apt    225               1                 45  2019-05-21   
2     Private room    150               3                  0         NaN   
3  Entire home/apt     89               1                270  2019-07-05   
4  Entire home/apt     80              10                  9  2018-11-19   

   reviews_per_month  calculated_host_listings_count  availability_365  
0               0.21                               6               365  
1               0.38                               2               355  
2                NaN                               1               365  
3               4.64                               1               194  
4               0.10                               1                 0

### Dataset Rows & Columns count

In [ ]:
# Dataset Rows & Columns count
dataset.shape

(48895, 16)

### Dataset Information

In [ ]:
# Dataset Info
# Getting info about the datatype column wise dtypes: float64(3 column), int64(7 column), object(6 column)
# & non null values present in the data
# last_review is in object datatype, it should be in datetime dtype
dataset.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 48895 entries, 0 to 48894
Data columns (total 16 columns):
 #   Column                          Non-Null Count  Dtype  
---  ------                          --------------  -----  
 0   id                              48895 non-null  int64  
 1   name                            48879 non-null  object 
 2   host_id                         48895 non-null  int64  
 3   host_name                       48874 non-null  object 
 4   neighbourhood_group             48895 non-null  object 
 5   neighbourhood                   48895 non-null  object 
 6   latitude                        48895 non-null  float64
 7   longitude                       48895 non-null  float64
 8   room_type                       48895 non-null  object 
 9   price                           48895 non-null  int64  
 10  minimum_nights                  48895 non-null  int64  
 11  number_of_reviews               48895 non-null  int64  
 12  last_review                     

#### Duplicate Values

In [ ]:
# Dataset Duplicate Value Count
# dataset.iloc[:,1:] have been used to ignore the unique id column values

print("Number of duplicate rows :",dataset.iloc[:,1:].duplicated().sum())

dataset.drop_duplicates(inplace=True)

Number of duplicate rows : 0


In [ ]:
# Dataset Duplicate Value Count
# check host id
duplicates = dataset["host_id"].duplicated()
dataset[duplicates]

id                                             name    host_id  \
22         8025                         CBG Helps Haiti Room#2.5      22486   
23         8110                            CBG Helps Haiti Rm #2      22486   
33         9783                              back room/bunk beds      32294   
35        10962  Lovely room 2 & garden; Best area, Legal rental       9744   
39        12048              LowerEastSide apt share shortterm 1       7549   
...         ...                                              ...        ...   
48887  36483152  Garden Jewel Apartment in Williamsburg New York  208514239   
48889  36484363                               QUIT PRIVATE HOUSE  107716952   
48890  36484665  Charming one bedroom - newly renovated rowhouse    8232441   
48891  36485057    Affordable room in Bushwick/East Williamsburg    6570630   
48893  36485609             43rd St. Time Square-cozy single bed   30985759   

             host_name neighbourhood_group       neighbourhood  latitude  \
22               Lisel            Brooklyn          Park Slope  40.67989   
23               Lisel            Brooklyn          Park Slope  40.68001   
33     Ssameer Or Trip           Manhattan              Harlem  40.82130   
35              Laurie            Brooklyn         South Slope  40.66869   
39                 Ben           Manhattan     Lower East Side  40.71401   
...                ...                 ...                 ...       ...   
48887            Melki            Brooklyn        Williamsburg  40.71232   
48889          Michael              Queens             Jamaica  40.69137   
48890          Sabrina            Brooklyn  Bedford-Stuyvesant  40.67853   
48891          Marisol            Brooklyn            Bushwick  40.70184   
48893              Taz           Manhattan      Hell's Kitchen  40.75751   

       longitude        room_type  price  minimum_nights  number_of_reviews  \
22     -73.97798     Private room     80               1                 39   
23     -73.97865     Private room    110               2                 71   
33     -73.95318     Private room     50               3                273   
35     -73.98780     Private room     89               4                168   
39     -73.98917      Shared room     40               1                214   
...          ...              ...    ...             ...                ...   
48887  -73.94220  Entire home/apt    170               1                  0   
48889  -73.80844     Private room     65               1                  0   
48890  -73.94995     Private room     70               2                  0   
48891  -73.93317     Private room     40               4                  0   
48893  -73.99112      Shared room     55               1                  0   

      last_review  reviews_per_month  calculated_host_listings_count  \
22     2019-01-01               0.37                               6   
23     2019-07-02               0.61                               6   
33     2019-07-01               2.37                               3   
35     2019-06-21               1.41                               3   
39     2019-07-05               1.81                               4   
...           ...                ...                             ...   
48887         NaN                NaN                               3   
48889         NaN                NaN                               2   
48890         NaN                NaN                               2   
48891         NaN                NaN                               2   
48893         NaN                NaN                               6   

       availability_365  
22                  364  
23                  304  
33                  359  
35                  340  
39                  188  
...                 ...  
48887               365  
48889               163  
48890                 9  
48891                36  
48893                 2  

[11438 rows x 16 columns

In [ ]:
# Dataset Duplicate Value Count
# check name of the listing
duplicates1 = dataset["name"].duplicated()
dataset[duplicates1]

id                                               name    host_id  \
330       81739                  Loft w/ Terrace @ Box House Hotel     417504   
339       84010                               Superior @ Box House     417504   
580      219818                 ✿✿✿ COUNTRY COTTAGE IN THE CITY✿✿✿    1138692   
661      250537                                The Lenox in Harlem    1313306   
669      253471                   Loft Suite @ The Box House Hotel     417504   
...         ...                                                ...        ...   
48684  36382847                                       Comfort home  266211707   
48735  36412461  Sunny, Cozy, Private Room In The Heart of Bush...  147515897   
48759  36420404                                    Home Sweet Home  273656890   
48791  36427922                                Home away from home  238163900   
48826  36449743                                  Brooklyn's finest   66084717   

                 host_name neighbourhood_group    neighbourhood  latitude  \
330    The Box House Hotel            Brooklyn       Greenpoint  40.73842   
339    The Box House Hotel            Brooklyn       Greenpoint  40.73813   
580           Keera (Jena)           Manhattan  Lower East Side  40.71892   
661                 Yvette           Manhattan           Harlem  40.81122   
669    The Box House Hotel            Brooklyn       Greenpoint  40.73641   
...                    ...                 ...              ...       ...   
48684                  Yan            Brooklyn      Sunset Park  40.64439   
48735               Flávia            Brooklyn         Bushwick  40.70366   
48759                Liana           Manhattan      East Harlem  40.79266   
48791                 Lucy              Queens  Cambria Heights  40.68557   
48826                  Tim            Brooklyn    East Flatbush  40.65170   

       longitude        room_type  price  minimum_nights  number_of_reviews  \
330    -73.95312     Private room    249               3                  2   
339    -73.95394     Private room    179               3                 13   
580    -73.98401  Entire home/apt    199               1                 14   
661    -73.94279  Entire home/apt    400               5                  0   
669    -73.95330  Entire home/apt    199               3                 24   
...          ...              ...    ...             ...                ...   
48684  -74.01816     Private room    185               1                  0   
48735  -73.92728     Private room     84               3                  0   
48759  -73.94740     Private room     50               1                  0   
48791  -73.72731     Private room     50               3                  0   
48826  -73.92580  Entire home/apt    200               3                  0   

      last_review  reviews_per_month  calculated_host_listings_count  \
330    2011-05-12               0.02                              28   
339    2019-06-27               0.14                              28   
580    2015-07-20               0.15                               2   
661           NaN                NaN                               2   
669    2018-11-06               0.32                              28   
...           ...                ...                             ...   
48684         NaN                NaN                               2   
48735         NaN                NaN                               1   
48759         NaN                NaN                               1   
48791         NaN                NaN                               1   
48826         NaN                NaN                               2   

       availability_365  
330                  60  
339                  81  
580                 158  
661                 365  
669                  84  
...                 ...  
48684               177  
48735                28  
48759                81  
48791               176  
48826          

#### Missing Values/Null Values

In [ ]:
# Missing Values/Null Values Count
dataset.isnull().sum()

id                                    0
name                                 16
host_id                               0
host_name                            21
neighbourhood_group                   0
neighbourhood                         0
latitude                              0
longitude                             0
room_type                             0
price                                 0
minimum_nights                        0
number_of_reviews                     0
last_review                       10052
reviews_per_month                 10052
calculated_host_listings_count        0
availability_365                      0
dtype: int64

In [ ]:
# Visualizing the missing values


### What did you know about your dataset?

Here we are having the datset of Air BNB 2019, we have to analyse the data and discover insights to provide better understanding of the requirements & facilitate decision making and give more appropriate & innovative services to our clients.

There are no duplicate rows in our dataset, duplicate host id values represents that individual host can post different rooms or apartment.

While duplicate name values represents that the listing could be a hotel or house having many different rooms availability.

The data is having 48895 rows, 16 columns. There are some null values in name, host_name,last_review and reviews_per_month column & non null values present in the data.

Last_review is in object datatype, while should be in datetime dtype

The last_review and reviews_per_month column may have null values because of "0" number_of_reviews at that destination .

## ***2. Understanding Your Variables***

In [ ]:
# Dataset Columns
dataset.columns

Index(['id', 'name', 'host_id', 'host_name', 'neighbourhood_group',
       'neighbourhood', 'latitude', 'longitude', 'room_type', 'price',
       'minimum_nights', 'number_of_reviews', 'last_review',
       'reviews_per_month', 'calculated_host_listings_count',
       'availability_365'],
      dtype='object')

In [ ]:
# Dataset Describe
dataset.describe()

id       host_id      latitude     longitude         price  \
count  4.889500e+04  4.889500e+04  48895.000000  48895.000000  48895.000000   
mean   1.901714e+07  6.762001e+07     40.728949    -73.952170    152.720687   
std    1.098311e+07  7.861097e+07      0.054530      0.046157    240.154170   
min    2.539000e+03  2.438000e+03     40.499790    -74.244420      0.000000   
25%    9.471945e+06  7.822033e+06     40.690100    -73.983070     69.000000   
50%    1.967728e+07  3.079382e+07     40.723070    -73.955680    106.000000   
75%    2.915218e+07  1.074344e+08     40.763115    -73.936275    175.000000   
max    3.648724e+07  2.743213e+08     40.913060    -73.712990  10000.000000   

       minimum_nights  number_of_reviews  reviews_per_month  \
count    48895.000000       48895.000000       38843.000000   
mean         7.029962          23.274466           1.373221   
std         20.510550          44.550582           1.680442   
min          1.000000           0.000000           0.010000   
25%          1.000000           1.000000           0.190000   
50%          3.000000           5.000000           0.720000   
75%          5.000000          24.000000           2.020000   
max       1250.000000         629.000000          58.500000   

       calculated_host_listings_count  availability_365  
count                    48895.000000      48895.000000  
mean                         7.143982        112.781327  
std                         32.952519        131.622289  
min                          1.000000          0.000000  
25%                          1.000000          0.000000  
50%                          1.000000         45.000000  
75%                          2.000000        227.000000  
max                        327.000000        365.000000

### Variables Description

1. Minimum_nights 1250 and minimum price 0 almost not possible

2. 327 listing from a single host needs to be analysed

3. 48895 Row, 16  Column in a dataset

4. Categorical data : host name, neighbourhood_group, neighbourhood, room_type

5. Data Has int, float and object data type

6. Id : It is unique for each bookings, numeric values, data is not repeatable.

7. name : How is the room. Ex : Clean & quiet apt home by the park, this can be neglected.

8. host_id : ID of the host, Data is repeatable, Numerical data .

9. host_name : Name of the host. This column is neglected, Categorical data .

10. neighbourhood_group : It is a location borough (a town that has its own goverment ). Brooklyn,Manhattan, Queens, Bronx, Staten Island .

11. Neighbourhood: Area of the location inside neighborhood group.

12. Latitude : Numerical values , Position of the location.

13. Longitude :Numerical values , Position of the location.

14. room_type : Private, Shared or Entire/Home, Categorical value.

15. price : Cost of the room, Numerical value.

16. minimum_nights : How many days compulsory to stay.

17. number_of_reviews : Total number of review for this host

18. last_review : Give information regarding last reviews is given in date.

19. reviews_per_month : Total number of reviews / Days in months

20. calculated_host_listings_count : The total number of apartments and bedrooms referred to the same landlord.

21. availability_365 : Days it is available in a year

22. The last_review and reviews_per_month column may have null values because of "0" number_of_reviews at that destination .

### Check Unique Values for each variable.

In [ ]:
# Check Unique Values for each variable.

unique_counts = dataset.nunique()
print(unique_counts)

id                                48895
name                              47905
host_id                           37457
host_name                         11452
neighbourhood_group                   5
neighbourhood                       221
latitude                          19048
longitude                         14718
room_type                             3
price                               674
minimum_nights                      109
number_of_reviews                   394
last_review                        1764
reviews_per_month                   937
calculated_host_listings_count       47
availability_365                    366
dtype: int64


## 3. ***Data Wrangling***

### Data Wrangling Code

In [ ]:
# Write your code to make your dataset analysis ready.
# Drop duplicate rows
dataset.drop_duplicates(inplace=True)
print("Number of duplicate rows :",dataset.duplicated().sum())

Number of duplicate rows : 0


In [ ]:
# All the rows having Nan values in name column
dataset[dataset["name"].isnull()== True]

# Removing Null values in name column
dataset.dropna(subset=['name'], inplace =True)

In [ ]:
# Replacing null values that are in host_name column
dataset["host_name"].fillna("Unknown host", inplace = True)

In [ ]:
# Replacing null values that are in column last_review and reviews_per_month column
# number_of_reviews = 0, could be the reason behind null values here.
dataset.fillna(0, inplace = True)


In [ ]:
# Check null values in dataset
dataset.isnull().sum()

id                                0
name                              0
host_id                           0
host_name                         0
neighbourhood_group               0
neighbourhood                     0
latitude                          0
longitude                         0
room_type                         0
price                             0
minimum_nights                    0
number_of_reviews                 0
last_review                       0
reviews_per_month                 0
calculated_host_listings_count    0
availability_365                  0
dtype: int64

In [ ]:
# Updated shape of the dataset
# 16 rows that have Null values in name column have been dropped
dataset.shape

(48879, 16)

In [ ]:
# Renaming some of the column names
dataset.rename(columns = {"calculated_host_listings_count":"Total Count","neighbourhood":"Area","neighbourhood_group":"location"}, inplace = True)
dataset.columns

Index(['id', 'name', 'host_id', 'host_name', 'location', 'Area', 'latitude',
       'longitude', 'room_type', 'price', 'minimum_nights',
       'number_of_reviews', 'last_review', 'reviews_per_month', 'Total Count',
       'availability_365'],
      dtype='object')

In [ ]:
# Seems that a lot of listing are not available right now.
print("Number of listing having 0 availability : ",dataset[dataset["availability_365"] == 0]["availability_365"].count())
dataset[dataset["availability_365"] == 0].head()

Number of listing having 0 availability :  17521


id                                              name  host_id  \
4   5022  Entire Apt: Spacious Studio/Loft by central park     7192   
6   5121                                   BlissArtsSpace!     7356   
8   5203                Cozy Clean Guest Room - Family Apt     7490   
14  6090                     West Village Nest - Superhost    11975   
20  7801                  Sweet and Spacious Brooklyn Loft    21207   

    host_name neighbourhood_group       neighbourhood  latitude  longitude  \
4       Laura           Manhattan         East Harlem  40.79851  -73.94399   
6       Garon            Brooklyn  Bedford-Stuyvesant  40.68688  -73.95596   
8   MaryEllen           Manhattan     Upper West Side  40.80178  -73.96723   
14      Alina           Manhattan        West Village  40.73530  -74.00525   
20      Chaya            Brooklyn        Williamsburg  40.71842  -73.95718   

          room_type  price  minimum_nights  number_of_reviews last_review  \
4   Entire home/apt     80              10                  9  2018-11-19   
6      Private room     60              45                 49  2017-10-05   
8      Private room     79               2                118  2017-07-21   
14  Entire home/apt    120              90                 27  2018-10-31   
20  Entire home/apt    299               3                  9  2011-12-28   

    reviews_per_month  calculated_host_listings_count  availability_365  
4                0.10                               1                 0  
6                0.40                               1                 0  
8                0.99                               1                 0  
14               0.22                               1                 0  
20               0.07                               1                 0

Different Unique values in host_name and host_id

In [ ]:
# Different Unique values in host_name and host_id
dataset[["host_name"]].value_counts()

host_name   
Michael         417
David           403
Sonder (NYC)    327
John            294
Alex            279
               ... 
Jerbean           1
Jerald            1
Jeonghoon         1
Jeny              1
현선                1
Length: 11451, dtype: int64

In [ ]:
dataset[["host_id"]].value_counts()

host_id  
219517861    327
107434423    232
30283594     121
137358866    103
16098958      96
            ... 
13538150       1
13535952       1
13533446       1
13532838       1
274321313      1
Length: 37443, dtype: int64

In [ ]:
# Procuring host_id of top host_name
# It can be observed that same host name can have different host_id

dataset[(dataset['host_name'] =="Michael")][["id","host_name","host_id"]]

id host_name    host_id
151       36121   Michael      62165
412      138216   Michael     674970
519      189135   Michael     179020
616      233189   Michael    1220414
811      289037   Michael     632334
...         ...       ...        ...
48378  36228909   Michael  272485928
48531  36310247   Michael   25498509
48786  36426788   Michael  266645207
48830  36450318   Michael  151831356
48889  36484363   Michael  107716952

[417 rows x 3 columns]

In [ ]:
# Number of rows having price < or = 0
d = dataset[(dataset['price'] <=0)].shape
print(f"Number of rows having price 0 or below : {d[0]}")

Number of rows having price 0 or below : 11


In [ ]:
# Removing the rows having price = 0
dataset.drop(dataset.index[(dataset['price'] <=0)],axis=0,inplace = True)
dataset.describe()

id       host_id      latitude     longitude         price  \
count  4.886800e+04  4.886800e+04  48868.000000  48868.000000  48868.000000   
mean   1.902027e+07  6.763215e+07     40.728948    -73.952174    152.756732   
std    1.098403e+07  7.862640e+07      0.054531      0.046162    240.202905   
min    2.539000e+03  2.438000e+03     40.499790    -74.244420     10.000000   
25%    9.473608e+06  7.812758e+06     40.690090    -73.983072     69.000000   
50%    1.967809e+07  3.079133e+07     40.723085    -73.955690    106.000000   
75%    2.915659e+07  1.074344e+08     40.763120    -73.936297    175.000000   
max    3.648724e+07  2.743213e+08     40.913060    -73.712990  10000.000000   

       minimum_nights  number_of_reviews  reviews_per_month  \
count    48868.000000       48868.000000       48868.000000   
mean         7.010948          23.278730           1.091138   
std         20.017604          44.556899           1.597363   
min          1.000000           0.000000           0.000000   
25%          1.000000           1.000000           0.040000   
50%          3.000000           5.000000           0.370000   
75%          5.000000          24.000000           1.580000   
max       1250.000000         629.000000          58.500000   

       calculated_host_listings_count  availability_365  
count                    48868.000000      48868.000000  
mean                         7.146538        112.787161  
std                         32.961406        131.618305  
min                          1.000000          0.000000  
25%                          1.000000          0.000000  
50%                          1.000000         45.000000  
75%                          2.000000        227.000000  
max                        327.000000        365.000000

In [ ]:
# Updated shape of the dataset
# 11 rows that have price = 0 have been dropped
dataset.shape

(48868, 16)

In [ ]:
# Checking the minimum_nights values in descending order
m = dataset.sort_values("minimum_nights",ascending = False)[["minimum_nights","price"]]
m[m["minimum_nights"]>365]

minimum_nights  price
5767             1250    180
38664             999    110
26341             999     79
13404             999     99
11193             500     50
47620             500    140
7355              500    134
14285             500     45
8014              500     75
10829             480    199
34487             400     50
1305              370    115
15946             366    250

In [ ]:
# Updation of last_review column to datetime dtype
# change last review datatype
dataset['last_review'] = pd.to_datetime(dataset['last_review'])
dataset.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 48868 entries, 0 to 48894
Data columns (total 16 columns):
 #   Column             Non-Null Count  Dtype         
---  ------             --------------  -----         
 0   id                 48868 non-null  int64         
 1   name               48868 non-null  object        
 2   host_id            48868 non-null  int64         
 3   host_name          48868 non-null  object        
 4   location           48868 non-null  object        
 5   Area               48868 non-null  object        
 6   latitude           48868 non-null  float64       
 7   longitude          48868 non-null  float64       
 8   room_type          48868 non-null  object        
 9   price              48868 non-null  int64         
 10  minimum_nights     48868 non-null  int64         
 11  number_of_reviews  48868 non-null  int64         
 12  last_review        48868 non-null  datetime64[ns]
 13  reviews_per_month  48868 non-null  float64       
 14  Total 

In [ ]:
# question 1. Number of listing location wise
dataset["location"].value_counts()

Manhattan        21651
Brooklyn         20089
Queens            5666
Bronx             1089
Staten Island      373
Name: location, dtype: int64

In [ ]:
# question 2. Number of listing Area wise
dataset["Area"].value_counts().head(20)

Williamsburg          3917
Bedford-Stuyvesant    3709
Harlem                2657
Bushwick              2460
Upper West Side       1970
Hell's Kitchen        1957
East Village          1852
Upper East Side       1798
Crown Heights         1564
Midtown               1545
East Harlem           1117
Greenpoint            1114
Chelsea               1113
Lower East Side        911
Astoria                900
Washington Heights     898
West Village           768
Financial District     744
Flatbush               621
Clinton Hill           572
Name: Area, dtype: int64

In [ ]:
# question 3. Top hosts having multiple listing on airbnb
dataset[["host_id","host_name"]].value_counts().head(15)

host_id    host_name        
219517861  Sonder (NYC)         327
107434423  Blueground           232
30283594   Kara                 121
137358866  Kazuya               103
16098958   Jeremy & Laura        96
12243051   Sonder                96
61391963   Corporate Housing     91
22541573   Ken                   87
200380610  Pranjal               65
7503643    Vida                  52
1475015    Mike                  52
120762452  Stanley               50
2856748    Ruchi                 49
205031545  Red Awning            49
190921808  John                  47
dtype: int64

In [ ]:
# question 4 distribution of price curve line curve
dataset[["price"]].value_counts()

price
100      2050
150      2046
50       1533
60       1458
200      1400
         ... 
342         1
898         1
338         1
428         1
506         1
Length: 673, dtype: int64

In [ ]:
# question 5 distribution of minimum_nights box plot
dataset[["minimum_nights"]].value_counts()

minimum_nights
1                 12709
2                 11693
3                  7996
30                 3758
4                  3299
                  ...  
175                   1
182                   1
183                   1
184                   1
1250                  1
Length: 108, dtype: int64

# Advance Data Wrangling

In [ ]:
# check the different one project

In [ ]:
# question . Number of listing location wise by room_type
# Made 2 datasets
# and merged with id having values location by room_type

df1 = dataset["id","location"]
df2 = dataset["id","room_type"]

dataset[["location"]].value_counts()

Manhattan        21651
Brooklyn         20089
Queens            5666
Bronx             1089
Staten Island      373
Name: location, dtype: int64

In [ ]:
# question
dataset[dataset["availability_365"] == 0].head(2)

id                                              name  host_id host_name  \
4  5022  Entire Apt: Spacious Studio/Loft by central park     7192     Laura   
6  5121                                   BlissArtsSpace!     7356     Garon   

    location                Area  latitude  longitude        room_type  price  \
4  Manhattan         East Harlem  40.79851  -73.94399  Entire home/apt     80   
6   Brooklyn  Bedford-Stuyvesant  40.68688  -73.95596     Private room     60   

   minimum_nights  number_of_reviews last_review  reviews_per_month  \
4              10                  9  2018-11-19                0.1   
6              45                 49  2017-10-05                0.4   

   Total Count  availability_365  
4            1                 0  
6            1                 0

### What all manipulations have you done and insights you found?

**Manipulations**

Duplicate rows are deleted. Null values are removed or replaced with zero.

 Here we can analyse that host_name could be same but the host_id could be different, host name is insignificant.

Some column names have been changed.

Price having 0 values in rows have been removed.

Updation of last_review column to datetime dtype

Data is all set for distribution, analysis and visualization.


---




**Insights**

Listings Minimum and 25% (Q1) minimum_nights is till 1 night stay and 75%(Q3) of the rooms is till 5 nights. Giving insight that 25% listing need 1 minimum night stay for booking while till 75% listing do have minimum requirement of 5 night stay. Rest 25% having criteria between 5 to 1250 nights.

Total count of listing per host is almost in range of 0 to 2.

A lot of listing do not have availability for booking. More analysis required.

More analysis and visualization below

## ***4. Data Vizualization, Storytelling & Experimenting with charts : Understand the relationships between variables***

#### Chart - 1

In [ ]:
# Chart - 1 visualization code

##### 1. Why did you pick the specific chart?

Answer Here.

##### 2. What is/are the insight(s) found from the chart?

Answer Here

##### 3. Will the gained insights help creating a positive business impact?
Are there any insights that lead to negative growth? Justify with specific reason.

Answer Here

#### Chart - 2

In [ ]:
# Chart - 2 visualization code

##### 1. Why did you pick the specific chart?

Answer Here.

##### 2. What is/are the insight(s) found from the chart?

Answer Here

##### 3. Will the gained insights help creating a positive business impact?
Are there any insights that lead to negative growth? Justify with specific reason.

Answer Here

#### Chart - 3

In [ ]:
# Chart - 3 visualization code

##### 1. Why did you pick the specific chart?

Answer Here.

##### 2. What is/are the insight(s) found from the chart?

Answer Here

##### 3. Will the gained insights help creating a positive business impact?
Are there any insights that lead to negative growth? Justify with specific reason.

Answer Here

#### Chart - 4

In [ ]:
# Chart - 4 visualization code

##### 1. Why did you pick the specific chart?

Answer Here.

##### 2. What is/are the insight(s) found from the chart?

Answer Here

##### 3. Will the gained insights help creating a positive business impact?
Are there any insights that lead to negative growth? Justify with specific reason.

Answer Here

#### Chart - 5

In [ ]:
# Chart - 5 visualization code

##### 1. Why did you pick the specific chart?

Answer Here.

##### 2. What is/are the insight(s) found from the chart?

Answer Here

##### 3. Will the gained insights help creating a positive business impact?
Are there any insights that lead to negative growth? Justify with specific reason.

Answer Here

#### Chart - 6

In [ ]:
# Chart - 6 visualization code

##### 1. Why did you pick the specific chart?

Answer Here.

##### 2. What is/are the insight(s) found from the chart?

Answer Here

##### 3. Will the gained insights help creating a positive business impact?
Are there any insights that lead to negative growth? Justify with specific reason.

Answer Here

#### Chart - 7

In [ ]:
# Chart - 7 visualization code

##### 1. Why did you pick the specific chart?

Answer Here.

##### 2. What is/are the insight(s) found from the chart?

Answer Here

##### 3. Will the gained insights help creating a positive business impact?
Are there any insights that lead to negative growth? Justify with specific reason.

Answer Here

#### Chart - 8

In [ ]:
# Chart - 8 visualization code

##### 1. Why did you pick the specific chart?

Answer Here.

##### 2. What is/are the insight(s) found from the chart?

Answer Here

##### 3. Will the gained insights help creating a positive business impact?
Are there any insights that lead to negative growth? Justify with specific reason.

Answer Here

#### Chart - 9

In [ ]:
# Chart - 9 visualization code

##### 1. Why did you pick the specific chart?

Answer Here.

##### 2. What is/are the insight(s) found from the chart?

Answer Here

##### 3. Will the gained insights help creating a positive business impact?
Are there any insights that lead to negative growth? Justify with specific reason.

Answer Here

#### Chart - 10

In [ ]:
# Chart - 10 visualization code

##### 1. Why did you pick the specific chart?

Answer Here.

##### 2. What is/are the insight(s) found from the chart?

Answer Here

##### 3. Will the gained insights help creating a positive business impact?
Are there any insights that lead to negative growth? Justify with specific reason.

Answer Here

#### Chart - 11

In [ ]:
# Chart - 11 visualization code

##### 1. Why did you pick the specific chart?

Answer Here.

##### 2. What is/are the insight(s) found from the chart?

Answer Here

##### 3. Will the gained insights help creating a positive business impact?
Are there any insights that lead to negative growth? Justify with specific reason.

Answer Here

#### Chart - 12

In [ ]:
# Chart - 12 visualization code

##### 1. Why did you pick the specific chart?

Answer Here.

##### 2. What is/are the insight(s) found from the chart?

Answer Here

##### 3. Will the gained insights help creating a positive business impact?
Are there any insights that lead to negative growth? Justify with specific reason.

Answer Here

#### Chart - 13

In [ ]:
# Chart - 13 visualization code

##### 1. Why did you pick the specific chart?

Answer Here.

##### 2. What is/are the insight(s) found from the chart?

Answer Here

##### 3. Will the gained insights help creating a positive business impact?
Are there any insights that lead to negative growth? Justify with specific reason.

Answer Here

#### Chart - 14 - Correlation Heatmap

In [ ]:
# Correlation Heatmap visualization code

##### 1. Why did you pick the specific chart?

Answer Here.

##### 2. What is/are the insight(s) found from the chart?

Answer Here

#### Chart - 15 - Pair Plot

In [ ]:
# Pair Plot visualization code

##### 1. Why did you pick the specific chart?

Answer Here.

##### 2. What is/are the insight(s) found from the chart?

Answer Here

## **5. Solution to Business Objective**

#### What do you suggest the client to achieve Business Objective ?
Explain Briefly.

Answer Here.

# **Conclusion**

Write the conclusion here.

### ***Hurrah! You have successfully completed your EDA Capstone Project !!!***